src/main/java/mekanism/common/item/gear/ItemMekaTool.java

```diff
@@ -223,6 +235,31 @@ public boolean hurtEnemy(@Nonnull ItemStack stack, @Nonnull LivingEntity target,
         return super.hurtEnemy(stack, target, attacker);
     }
 
+    @Override
+    public Map<BlockPos, BlockState> getBlastedBlocks(Level world, Player player, ItemStack stack, BlockPos pos, BlockState state) {
+        //Setup initial set for blasting
+        if (!player.isShiftKeyDown()) {
+            IModule<ModuleBlastingUnit> blastingUnit = getModule(stack, MekanismModules.BLASTING_UNIT);
+            if (blastingUnit != null && blastingUnit.isEnabled()) {
+                int radius = blastingUnit.getCustomInstance().getBlastRadius();
+                if (radius > 0 && IBlastingItem.canBlastBlock(world, pos, state)) {
+                    Direction direction = getPlayerPOVHitResult(world, player, ClipContext.Fluid.NONE).getDirection();
+                    return IBlastingItem.findPositions(world, pos, direction, radius);
+                }
+            }
+        }
+        return Collections.emptyMap();
+    }
+
+    private Object2IntMap<BlockPos> getVeinedBlocks(Level world, ItemStack stack, Map<BlockPos, BlockState> blocks, Object2BooleanMap<Block> oreTracker) {
+        IModule<ModuleVeinMiningUnit> veinMiningUnit = getModule(stack, MekanismModules.VEIN_MINING_UNIT);
+        if (veinMiningUnit != null && veinMiningUnit.isEnabled()) {
+            ModuleVeinMiningUnit customInstance = veinMiningUnit.getCustomInstance();
+            return ModuleVeinMiningUnit.findPositions(world, blocks, customInstance.isExtended() ? customInstance.getExcavationRange() : 0, oreTracker);
```

Comment:
Thinking about it we may as well simplify the logic here like we do in blasted blocks where if it is on extended BUT the excavation range is zero, then we can just skip the findPositions logic.

Reply:
It still might find positions for ores, we could check the oreTracker to see if any are ores though if you want.

Reply:
Ah, because of the fact of how blasting and veining interact. Hmm I will try to think about this some more.

Reply:
well even with blasting off you'd still have to go to findPositions for the target block in case it is an ore, I think it used to have the check before going to findPositions and we could do basically the same thing by checking the oreTracker.any but I don't think we're looking at a significant performance difference over the traversal just ending quickly.

Reply:
True, I was forgetting radius zero would imply that extended is off not that it can't vein mine at all.